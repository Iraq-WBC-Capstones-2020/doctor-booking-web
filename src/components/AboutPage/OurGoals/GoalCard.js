import React from 'react';
import { Card } from 'react-bootstrap';
import PropTypes from 'prop-types';
function GoalCard(props) {
  GoalCard.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  };
  return (
    <Card className="goal-card">
      <Card.Img className="card-image mt-4" variant="top" src={props.img} />
      <Card.Body className="mt-3 mb-3">
        <Card.Title className="text-center card-title">
          {props.title}
        </Card.Title>
        <Card.Text className="text-center card-text px-3 pt-3">
          {props.text}
        </Card.Text>
      </Card.Body>
    </Card>
  );
}

export default GoalCard;
