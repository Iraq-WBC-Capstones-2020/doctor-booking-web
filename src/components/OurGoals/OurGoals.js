import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';
function OurGoals() {
  return (
    <Container>
      <Row>
        <Col md="4"></Col>
        <Col md="4"></Col>
        <Col md="4"></Col>
      </Row>
    </Container>
  );
}

export default OurGoals;
