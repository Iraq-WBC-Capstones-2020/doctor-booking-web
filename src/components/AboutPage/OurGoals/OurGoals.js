import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import GoalCard from './GoalCard';
import numberOneImg from './imgs/no1.svg';
import locationImg from './imgs/location.svg';
import awarenessImg from './imgs/awareness.svg';
import { useTranslation } from 'react-i18next';

function OurGoals() {
  const { t } = useTranslation(['translation', 'validation']);

  return (
    <Container className="abSection">
      <Row className="justify-content-center">
        <h2 className="section-title">
          {t('validation:aboutPage.ourGoals1.headLine')}
        </h2>
      </Row>
      <Row className="justify-content-between">
        <Col className="justify-content-center d-flex mt-5" md="4">
          <GoalCard
            img={numberOneImg}
            title={t('validation:aboutPage.ourGoals1.card1.headLine')}
            text={t('validation:aboutPage.ourGoals1.card1.paragraph')}
          />
        </Col>
        <Col className="justify-content-center d-flex mt-5" md="4">
          <GoalCard
            img={locationImg}
            title={t('validation:aboutPage.ourGoals1.card2.headLine')}
            text={t('validation:aboutPage.ourGoals1.card2.paragraph')}
          />
        </Col>
        <Col className="justify-content-center d-flex mt-5" md="4">
          <GoalCard
            img={awarenessImg}
            title={t('validation:aboutPage.ourGoals1.card3.headLine')}
            text={t('validation:aboutPage.ourGoals1.card3.paragraph')}
          />
        </Col>
      </Row>
    </Container>
  );
}

export default OurGoals;
