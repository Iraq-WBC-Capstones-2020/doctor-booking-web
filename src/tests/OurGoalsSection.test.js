import React from 'react';
import { render } from '@testing-library/react';
import OurGoals from '../components/OurGoals/OurGoals';
import renderer from 'react-test-renderer';
test('our goals section is rendering ', () => {
  render(<OurGoals />);
});

test('our goals section renders correctly', () => {
  const tree = renderer.create(<OurGoals />).toJSON();
  expect(tree).toMatchSnapshot();
});
