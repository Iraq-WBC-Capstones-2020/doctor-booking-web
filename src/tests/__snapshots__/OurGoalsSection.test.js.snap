// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`our goals section renders correctly 1`] = `
<div
  className="abSection container"
>
  <div
    className="justify-content-center row"
  >
    <h2
      className="section-title"
    >
      Our Goals
    </h2>
  </div>
  <div
    className="justify-content-between row"
  >
    <div
      className="justify-content-center d-flex mt-5 col-md-4"
    >
      <div
        className="goal-card card"
      >
        <img
          className="card-img-top card-image mt-4"
          src="no1.svg"
        />
        <div
          className="mt-3 mb-3 card-body"
        >
          <div
            className="text-center card-title card-title h5"
          >
            1st
          </div>
          <p
            className="text-center card-text px-3 pt-3 card-text"
          >
            one of our goals is to be  best doctor booking website in Erbil
          </p>
        </div>
      </div>
    </div>
    <div
      className="justify-content-center d-flex mt-5 col-md-4"
    >
      <div
        className="goal-card card"
      >
        <img
          className="card-img-top card-image mt-4"
          src="location.svg"
        />
        <div
          className="mt-3 mb-3 card-body"
        >
          <div
            className="text-center card-title card-title h5"
          >
            Register all clinics
          </div>
          <p
            className="text-center card-text px-3 pt-3 card-text"
          >
            Register all the clinics in Erbil so we offer the users the desired flexibility to choose the doctor that the user wants
          </p>
        </div>
      </div>
    </div>
    <div
      className="justify-content-center d-flex mt-5 col-md-4"
    >
      <div
        className="goal-card card"
      >
        <img
          className="card-img-top card-image mt-4"
          src="awareness.svg"
        />
        <div
          className="mt-3 mb-3 card-body"
        >
          <div
            className="text-center card-title card-title h5"
          >
            Awareness
          </div>
          <p
            className="text-center card-text px-3 pt-3 card-text"
          >
            Reducnig the gatherings in clinics waiting rooms so we ensure the social distancing to be applied
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
